rest-request.go

```diff
@@ -58,17 +58,20 @@ type StatusMessage struct {
 // NewClient initializes client for interacting with an instance of Grafana server;
 // apiKeyOrBasicAuth accepts either 'username:password' basic authentication credentials,
 // or a Grafana API key
-func NewClient(apiURL, apiKeyOrBasicAuth string, client *http.Client) *Client {
+func NewClient(apiURL, apiKeyOrBasicAuth string, client *http.Client) (*Client, error) {
 	key := ""
 	basicAuth := strings.Contains(apiKeyOrBasicAuth, ":")
-	baseURL, _ := url.Parse(apiURL)
+	baseURL, err := url.Parse(apiURL)
+	if err != nil {
+		return nil, err
+	}
 	if !basicAuth {
 		key = fmt.Sprintf("Bearer %s", apiKeyOrBasicAuth)
 	} else {
 		parts := strings.Split(apiKeyOrBasicAuth, ":")
 		baseURL.User = url.UserPassword(parts[0], parts[1])
```

Comment:
Just noticed that we are missing some error handling here as well :eye: it would nice to get a proper error message if `apiKeyOrBasicAuth` is malformed instead of a panic